src/oemof_tabular_plugins/wefe/facades/crops.py

```diff
@@ -0,0 +1,1176 @@
+            else:
+                f_temp = 1
+            return f_temp
+
+        def heat(t_air, t_max, t_ext, **kwargs):
+            """Heat stress effect on plant growth (SIMPLE) [-]"""
+            if t_air <= t_max:
+                f_heat = 1
+            elif t_max < t_air <= t_ext:
+                f_heat = 1 - (t_air - t_max) / (t_ext - t_max)
+            else:
+                f_heat = 0
+            return f_heat
+
+        def soil_heat_flux(ghi, irr_w, **kwargs):
+            """Soil heat flux as fraction of incoming radiation (FAO56) [W/m²]"""
+            if ghi > 1e-5:
+                g = 0.1 * irr_w
+            else:
+                g = 0.5 * irr_w
+            return g
+
+        def potential_evapotranspiration(z, t_air, t_dp, w10, irr_w, g_w, **kwargs):
+            """Potential evapotranspiration for reference grass (FAO56) [mm]"""
+            cp_air = 1.013e-3  # specific heat at constant pressure [MJ/(kg °C)]
+            epsilon = 0.622  # ratio molecular weight of water vapour/dry air [-]
+            h_vap = 2.45  # latent heat of vaporization [MJ/kg]
+            rho_w = 1000  # density of water [kg/m³]
+            k_c = 1  # crop coefficient (FAO) [-]
+            irr = 86.4e-3 * irr_w  # W/m² to MJ/m²*day
+            g = 3.6e-3 * g_w  # W/m² to MJ/m²*day
+            w2 = (
+                w10 * 4.87 / np.log(672.58)
+            )  # wind speed at 2m above ground (FAO56) [m/s]
+            p = 101.3 * ((293 - 0.0065 * z) / 293) ** 5.26  # atmospheric pressure [kPa]
+            gamma = cp_air * p / (h_vap * epsilon)  # psychrometric constant
+            delta = (
+                4098
+                * (0.6108 * np.exp(17.27 * t_air / (t_air + 237.3)))
+                / (t_air + 237.3) ** 2
+            )  # slope of sat vap press curve
+
+            def vap_pressure(t):
+                """Water vapor saturation pressure at specific temperature [kPa]"""
+                e = 0.6108 * np.exp(17.27 * t / (t + 237.3))
+                return e
+
+            e_s = vap_pressure(t_air)  # saturation vapour pressure [kPa]
+            e_a = vap_pressure(
+                t_dp
+            )  # actual vapour pressure (sat vap press at dewpoint temp) [kPa]
+
+            et_0 = (
+                0.408 * delta * (irr - g)
+                + gamma * 900 / (t_air + 273) * w2 * (e_s - e_a)
+            ) / (
+                delta + gamma * 1.34 * w2
+            )  # [mm/m²*day]
+            et_p = k_c * et_0 / 24  # [mm/m²*h]
+            # q_et = et * rho_w * h_vap / 1000  # [W/m²]
+            return et_p
+
+        def runoff(p, rcn, **kwargs):
+            """Surface runoff of precipitation (ARID) [mm]"""
+            s = 25400 / rcn - 254
+            i_a = 0.2 * s
+            if p > i_a:
+                r = (p - i_a) ** 2 / (p + i_a - s) / 24
+            else:
+                r = 0.0
+            return r
+
+        def soil_water_balance(df, has_irrigation, awc, ddc, rzd, **kwargs):
```

Comment:
it is not very easy to keep track here of what is what, there are some timeseries which get implicitly created within this function